src/style/style.js

```diff
@@ -802,15 +802,15 @@ class Style extends Evented {
             this.fire(new ErrorEvent(new Error(`The sourceLayer parameter must be provided for vector source types.`)));
             return;
         }
-        if (feature.id == null || feature.id === "") {
-            this.fire(new ErrorEvent(new Error(`The feature id parameter must be provided.`)));
+        if (feature.id == null || feature.id < 0) {
+            this.fire(new ErrorEvent(new Error(`The feature id parameter must be provided and non-negative.`)));
```

Comment:
I think `Map#setFeatureState` should still accept string types that can be coerced to integers. This would allow what is otherwise a valid `id` for a GeoJSON feature to be input to the API without explicit conversion. 

Based on https://github.com/mapbox/mapbox-gl-js/pull/4581:

https://github.com/mapbox/mapbox-gl-js/blob/3b0a1a9c3fceaefe1b0cd6f2d36cf92138621715/src/source/geojson_wrapper.js#L38-L46

Reply:
@bfrengley that seems like a fix worth putting in as well.